exercises/concept/karls-languages/src/test/java/LanguageListTest.java

```diff
@@ -1,44 +1,32 @@
-import static org.assertj.core.api.Assertions.assertThat;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.Test;
 
-import org.junit.Test;
+import static org.assertj.core.api.Assertions.*;
 
 public class LanguageListTest {
 
     LanguageList languageList = new LanguageList();
 
     @Test
+    @Tag("task:1")
+    @DisplayName("The isEmpty method returns true when no language was added to the list")
     public void empty() {
         assertThat(languageList.isEmpty()).isTrue();
     }
 
     @Test
+    @Tag("task:2")
+    @DisplayName("The addLanguage method correctly adds a language to the list so that list is no more empty")
     public void nonEmpty() {
         languageList.addLanguage("Java");
 
         assertThat(languageList.isEmpty()).isFalse();
     }
 
     @Test
-    public void addOneLanguage() {
-        languageList.addLanguage("Java");
-
-        assertThat(languageList.containsLanguage("Java")).isTrue();
-        assertThat(languageList.containsLanguage("Python")).isFalse();
-    }
-
-    @Test
-    public void addMultipleLanguages() {
-        languageList.addLanguage("Java");
-        languageList.addLanguage("Ruby");
-        languageList.addLanguage("C++");
-
-        assertThat(languageList.containsLanguage("Java")).isTrue();
-        assertThat(languageList.containsLanguage("Ruby")).isTrue();
-        assertThat(languageList.containsLanguage("C++")).isTrue();
-        assertThat(languageList.containsLanguage("Python")).isFalse();
-    }
-
-    @Test
+    @Tag("task:3")
+    @DisplayName("The removeLanguage method correctly removes a language from the list")
```

Comment:
This test is a bit weird: it uses `containsLanguage` to check whether the language was removed, but this method is not yet implemented in task 3.

I would suggest refactoring the test method to only use methods implemented so far, by making it

1. Add a language
2. Remove that same language
3. Asserting that the list is empty again.